labs/sysdev-buildroot-stm32mp2/sysdev-buildroot-stm32mp2.tex

```diff
@@ -0,0 +1,577 @@
+  \item Set \code{Custom scripts to run after creating filesystem images} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/common/post-image.sh}
+  \item Set \code{Extra arguments passed to custom scripts} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/stm32mp2/genimage.cfg}
+  \end{itemize}
+    
+\item \code{Kernel}
+  \begin{itemize}
+  \item Enable \code{Linux Kernel}
+  \item Set \code{Kernel version} to \code{Custom tarball}
+  \item Set \code{URL of custom kernel tarball}to \code{$(call github,STMicroelectronics,linux)v6.6-stm32mp-r1.tar.gz}
+  \item Set \code{Custom kernel patches} to \code{board/bootlin/training/0001-Custom-DTS-for-Bootlin-lab.patch}
+  \item Set \code{Kernel configuration} to \code{Using a custom (def)config file})
+  \item Set \code{Configuration file path} to \code{board/bootlin/training/linux.config}
+  \item Select \code{Build a Device Tree Blob (DTB)}
+  \item Set \code{In-tree Device Tree Source file names} to \code{st/stm32mp257f-dk-custom}
+  \end{itemize}
+\item \code{Target packages}
+  \begin{itemize}
+  \item Keep \code{BusyBox} (default version) and keep the BusyBox
+    configuration proposed by Buildroot;
+  \item \code{Audio and video applications}
+    \begin{itemize}
+    \item Select \code{alsa-utils}, and in the submenu:
+    \begin{itemize}
+         \item Only keep \code{speaker-test}
+    \end{itemize}
+    \item Select \code{mpd}, and in the submenu:
+    \begin{itemize}
+         \item Keep only \code{alsa}, \code{vorbis} and \code{tcp sockets}
+    \end{itemize}
+    \item Select \code{mpd-mpc}.
+    \end{itemize}
+  \item \code{Hardware handling}
+    \begin{itemize}
+	 \item Select \code{evtest}\\
+	       This userspace application allows to test events from
+	       input devices. This way, we will be able to test the
+	       Nunchuk by getting details about which buttons were
+	       pressed.
+    \end{itemize}
+  \end{itemize}
+\item \code{Filesystem images}
+  \begin{itemize}
+  \item Select \code{ext2/3/4 root filesystem} and set to \code{ext4}
+  \item Set \code{exact size} to \code{200M}
+  \item Select \code{tar the root filesystem}
+  \end{itemize}
+\item \code{Bootloaders}
+  \begin{itemize}
+
+  \item Select \code{ARM Trusted Firmware (ATF)}
+  \begin{itemize}
+    \item Set \code{ATF - verssion} to \code{custom tarball }
+    \item Set \code{URL of custom ATF tarball} to \code{$(call github,STMicroelectronics,arm-trusted-firmware)v2.10-stm32mp-r1.tar.gz}
+    \item Set \code{ATF platform } to \code{stm32mp2}
+    \item Select \code{build fip image}
+    \item Set \code{U-Boot BL33 image name} to \code{u-boot-nodtb.bin}  
+    \item Set \code{Additional ATF build variables} to \code{STM32MP_LPDDR4_TYPE=1 DTB_FILE_NAME=stm32mp257f-dk.dtb STM32MP_SDMMC=1 BL32_EXTRA2= BL33_CFG=$(BINARIES_DIR)/u-boot.dtb}
+    \item Set \code{Binary boot images} to \code{*.bin *.stm32}
+    \item Select \code{Needs dtc}
+  \end{itemize}
```

Comment:
In the Buildroot lab on other platforms, we don't build the bootloaders. So the build of TF-A/OP-TEE/U-Boot can be dropped here.